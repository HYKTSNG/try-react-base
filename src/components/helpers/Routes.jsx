import React from "react"
import { HashRouter, Route, Switch } from "react-router-dom"
import { NotFound } from "src/components/pages/NotFound"
import { Root } from "src/components/pages/Root"
import { routePath } from "src/constants/routePath"

export const Routes = () => {
  return (
    <HashRouter>
      <Switch>
        <Route component={Root} exact path={routePath.root} />

        {/* No route */}
        <Route component={NotFound} path={routePath.root} />
        <Route component={NotFound} />
      </Switch>
    </HashRouter>
  )
}
