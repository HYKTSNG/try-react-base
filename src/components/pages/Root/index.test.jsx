import { render } from "@testing-library/react"
import React from "react"
import { Routes } from "src/components/helpers/Routes"

import { Root } from "./"

describe("snapshot testing", () => {
  test("render", () => {
    // ## Arrange ##
    // ## Act ##
    const { container } = render(
      <Routes>
        <Root />
      </Routes>
    )

    // ## Assert ##
    expect(container).toMatchSnapshot()
  })
})
