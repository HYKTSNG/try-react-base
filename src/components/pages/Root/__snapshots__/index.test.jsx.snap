// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing render 1`] = `
<div>
  <div>
    <div
      class="css-6w1szs-Root"
    >
      <a
        href="#/counter"
      >
        Counter App
      </a>
    </div>
    <div
      class="css-6w1szs-Root"
    >
      <a
        href="#/ticTacToe"
      >
        TicTacToe App
      </a>
    </div>
    <div
      class="css-6w1szs-Root"
    >
      <a
        href="#/todo"
      >
        Todo App
      </a>
    </div>
  </div>
</div>
`;
